src/node_i18n.cc

```diff
@@ -436,13 +436,25 @@ void ConverterObject::Create(const FunctionCallbackInfo<Value>& args) {
 void ConverterObject::Decode(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 
-  CHECK_GE(args.Length(), 3);  // Converter, Buffer, Flags
+  CHECK_GE(args.Length(), 4);  // Converter, Buffer, Flags, Encoding
 
   ConverterObject* converter;
   ASSIGN_OR_RETURN_UNWRAP(&converter, args[0].As<Object>());
+
+  if (!(args[1]->IsArrayBuffer() || args[1]->IsSharedArrayBuffer() ||
+        args[1]->IsArrayBufferView())) {
+    return node::THROW_ERR_INVALID_ARG_TYPE(
+        env->isolate(),
+        "The \"input\" argument must be an instance of SharedArrayBuffer, "
+        "ArrayBuffer or ArrayBufferView.");
+  }
+
   ArrayBufferViewContents<char> input(args[1]);
   int flags = args[2]->Uint32Value(env->context()).ToChecked();
 
+  CHECK(args[3]->IsString());
+  Local<String> fromEncoding = args[3].As<String>();
```

Comment:
```suggestion
  Local<String> from_encoding = args[3].As<String>();
```

(https://github.com/nodejs/node/blob/main/doc/contributing/cpp-style-guide.md#snake_case-for-local-variables-and-parameters)

Reply:
I updated the pull request with the changes. Thank you for the review! My reasoning was towards reducing the C++ bridge interaction. Previously, `isAnyArrayBuffer` and `isArrayBufferView` each called the C++ side.